mlflow/genai/evaluation/utils.py

```diff
@@ -55,6 +55,153 @@
 )
 
 
+def get_evaluation_data_size_and_type(data: "EvaluationDatasetTypes") -> dict[str, Any]:
+    """
+    Get the size and type of evaluation data for telemetry tracking.
+
+    Args:
+        data: The data object to analyze. Must be one of the types in EvaluationDatasetTypes:
+            - pd.DataFrame
+            - pyspark.sql.dataframe.DataFrame
+            - list[dict]
+            - list[Trace]
+            - genai.EvaluationDataset
+            - entities.EvaluationDataset
+
+    Returns:
+        Dictionary with 'eval_data_size' (number of rows) and 'eval_data_type' (string type name)
+    """
+    from mlflow.entities.evaluation_dataset import EvaluationDataset as EntityEvaluationDataset
+    from mlflow.genai.datasets import EvaluationDataset as ManagedEvaluationDataset
+
+    eval_data_type = None
+    eval_data_size = None
+
+    try:
+        if isinstance(data, ManagedEvaluationDataset):
+            eval_data_type = "genai.EvaluationDataset"
+            # Check if records are loaded to avoid triggering expensive load
+            if data.has_records():
+                df = data.to_df()
+                eval_data_size = len(df)
+            else:
+                # Records not loaded, don't trigger load for telemetry
+                eval_data_size = None
+        elif isinstance(data, EntityEvaluationDataset):
```

Comment:
This is basically the same as above except data type, can we unify them as `if isinstance(data, (xxx, xxx))`?